var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -15,19 +15,15 @@ class Vtk(CMakePackage):
     available software system for 3D computer graphics, image
     processing and visualization."""
 
-    homepage = "http://www.vtk.org"
+    homepage = "https://www.vtk.org"
     url = "https://www.vtk.org/files/release/9.0/VTK-9.0.0.tar.gz"
-    list_url = "http://www.vtk.org/download/"
+    list_url = "https://www.vtk.org/download/"
 
     maintainers = ["chuckatkins", "danlipsa"]
 
     version("9.2.2", sha256="1c5b0a2be71fac96ff4831af69e350f7a0ea3168981f790c000709dcf9121075")
     version("9.1.0", sha256="8fed42f4f8f1eb8083107b68eaa9ad71da07110161a3116ad807f43e5ca5ce96")
-    version(
-        "9.0.3",
-        sha256="bc3eb9625b2b8dbfecb6052a2ab091fc91405de4333b0ec68f3323815154ed8a",
-        preferred=True,
```

Comment:
This should still be preferred over the 9.0.0 and 9.0.1 releases, I think that is why it is here. Does the new concretizer prioritize highest possible version now? The old one didn't which is why this went in originally.

Reply:
The concretizer always has and always will prioritize the most recent stable release.

Reply:
If you notice that again, ping @alalazo and he can probably explain what's going on. I'm guessing it's trying to avoid rebuilding something that is already installed or trying to minimize dependencies by selecting an older version.